iree/modules/check/CMakeLists.txt

```diff
@@ -12,41 +12,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory(dialect)
+if(${IREE_BUILD_COMPILER})
```

Comment:
Add

```
# bazel_to_cmake: DO NOT EDIT
```

?

Other options (leaving for future work is okay):

* Put the conditionals in `iree/modules/CMakeLists.txt` and let these subdirectories be autogenerated
* Teach `bazel_to_cmake.py` about `IREE_BUILD_COMPILER`
* Refactor so `iree/modules/` does not depend on the compiler, put compiler components under `iree/compiler`